pkg/controller/account/iam.go

```diff
@@ -358,3 +361,79 @@ func getSigninToken(reqLogger logr.Logger, federationEndpointURL string, federat
 	return signinResponse, nil
 
 }
+
+func (r *ReconcileAccount) smartCreateAccessKey(reqLogger logr.Logger, client awsclient.Client, userName string, account *awsv1alpha1.Account) (*iam.CreateAccessKeyOutput, error) {
+
+	output := &iam.CreateAccessKeyOutput{}
+	var err error
+	var accessKeyID string
+
+	if account.Spec.IAMUserSecret == "" {
+		output, err = createOrRotateAccesskey(reqLogger, client, userName)
+		if err != nil {
+			return &iam.CreateAccessKeyOutput{}, err
+		}
+	} else {
+		iamSecret := &corev1.Secret{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: account.Spec.IAMUserSecret, Namespace: awsv1alpha1.AccountCrNamespace}, iamSecret)
+		if err != nil {
+			return &iam.CreateAccessKeyOutput{}, err
+		}
+
+		if encryptedKeyID, ok := iamSecret.Data["aws_access_key_id"]; ok {
+			accessKeyID = string(encryptedKeyID)
+		} else {
+			return &iam.CreateAccessKeyOutput{}, errors.New("SecretMissingAccessKey")
+		}
+
+		err = smartDeleteAccessKey(reqLogger, client, userName, accessKeyID)
+		if err != nil {
+			return &iam.CreateAccessKeyOutput{}, err
+		}
+
+		output, err = createOrRotateAccesskey(reqLogger, client, userName)
+		if err != nil {
+			return &iam.CreateAccessKeyOutput{}, err
+		}
+	}
+
+	return output, nil
+}
+
+func createOrRotateAccesskey(reqLogger logr.Logger, client awsclient.Client, userName string) (*iam.CreateAccessKeyOutput, error) {
+
+	output, err := CreateUserAccessKey(reqLogger, client, userName)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == "LimitExceeded" {
+				accessKeyList, err := client.ListAccessKeys(&iam.ListAccessKeysInput{})
+				if err != nil {
+					return nil, err
+				}
+				smartDeleteAccessKey(reqLogger, client, userName, *accessKeyList.AccessKeyMetadata[0].AccessKeyId)
+			}
+		}
+	}
+
+	return output, nil
+}
+
+func smartDeleteAccessKey(reqLogger logr.Logger, client awsclient.Client, userName string, accessKeyID string) error {
```

Comment:
This function should be just `deleteAccessKeys`

Reply:
It deletes the other access key than given. Don't remember why I exactly did that though.